src/lib/libXbSymbolDatabase.c

```diff
@@ -665,6 +671,14 @@ uint32_t XbSymbolDatabase_GenerateSectionFilter(const void* xb_header_addr, XbSD
                     break;
                 }
             }
+
+            // If section is not on the filter list, check if there is kernel thunk table within section to enforce include it.
+            if (count == count_check) {
+                if (kernel_thunk_addr >= sh_index->dwVirtualAddr && kernel_thunk_addr < (sh_index->dwVirtualAddr + sh_index->dwSizeofRaw)) {
+                    // Once found within range, then enforce add section to the filter.
+                    goto enforceAddToSectionFilter;
```

Comment:
The use of goto is generally considered a bad code smell; It's probably better to extract-method the entire block that needs to be performed (again), any required variables passed in as arguments.

Reply:
We do have goto in our codebase. However, it is a necessary to perform clean up properly. Now, in this pull request no longer have goto. 😃